src/module/item/spell/document.ts

```diff
@@ -789,25 +765,27 @@ class SpellPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Ite
 
         const { baseRank } = this;
         const heightened = castLevel - baseRank;
-        const rankLabel = (() => {
-            const type = this.isCantrip
-                ? localize("PF2E.TraitCantrip")
-                : localize(CONFIG.PF2E.spellCategories[this.system.category.value]);
-            return game.i18n.format("PF2E.ItemLevel", { type, level: castLevel });
-        })();
+        const rankLabel = createSpellRankLabel(this, castLevel);
 
         // Combine properties
         const area = this.area;
-        const properties = R.compact([
-            heightened ? game.i18n.format("PF2E.SpellLevelBase", { level: ordinalString(baseRank) }) : null,
-            heightened ? game.i18n.format("PF2E.SpellLevelHeightened", { heightened }) : null,
-            this.isRitual ? null : `${localize("PF2E.SpellComponentsLabel")}: ${this.components.value}`,
-            systemData.range.value ? `${localize("PF2E.SpellRangeLabel")}: ${systemData.range.value}` : null,
-            systemData.target.value ? `${localize("PF2E.SpellTargetLabel")}: ${systemData.target.value}` : null,
-            area ? game.i18n.format("PF2E.SpellArea", { area: area.label }) : null,
-            systemData.time.value ? `${localize("PF2E.SpellTimeLabel")}: ${systemData.time.value}` : null,
-            systemData.duration.value ? `${localize("PF2E.SpellDurationLabel")}: ${systemData.duration.value}` : null,
-        ]);
+        const properties = R.compact(
+            [
+                heightened ? game.i18n.format("PF2E.SpellLevelBase", { level: ordinalString(baseRank) }) : null,
+                heightened ? game.i18n.format("PF2E.SpellLevelHeightened", { heightened }) : null,
+                systemData.range.value
+                    ? `${game.i18n.localize("PF2E.SpellRangeLabel")}: ${systemData.range.value}`
+                    : null,
+                systemData.target.value
+                    ? `${game.i18n.localize("PF2E.SpellTargetLabel")}: ${systemData.target.value}`
+                    : null,
+                area ? game.i18n.format("PF2E.SpellArea", { area: area.label }) : null,
+                systemData.time.value ? `${game.i18n.localize("PF2E.SpellTimeLabel")}: ${systemData.time.value}` : null,
+                systemData.duration.value
+                    ? `${game.i18n.localize("PF2E.SpellDurationLabel")}: ${systemData.duration.value}`
+                    : null,
+            ].filter((p): p is string => p !== null),
```

Comment:
Why is this filter required? Isn't that the purpose of compact?

Reply:
That part isn't new

Reply:
It wasnt, but i removed it recently. This PR adds it back.

Might have come about from resolving conflicts then.